dask/array/tests/test_cupy_reductions.py

```diff
@@ -56,15 +56,15 @@ def test_nanarg_reductions(dfunc, func):
     assert_eq(dfunc(a), func(x))
     assert_eq(dfunc(a, 0), func(x, 0))
     with pytest.raises(ValueError):
-        with pytest.warns(None):  # All NaN axis
+        with pytest.warns(RuntimeWarning, match="All-NaN slice encountered"):
```

Comment:
Hmmm, this changes the meaning of the `with pytest.warns` block. Previously that was used just to silence warnings, but this changes it to assert that a warning happens (which we don't necessarily care that it does). Since the warning is raised by numpy (and not dask), if numpy changes their warning message this will break our test.

We could use `pytest.mark.filterwarnings` if you want to be explicit about the warning that we're ignoring, or one of the other patterns you've used with `warnings.catch_warnings` to silence all warnings in the block.

Reply:
Yeah I kind of changed my thinking as I was working through these. I'll fix make them use `catch_warnings` instead.

Reply:
Ok I switched them to `warnings.catch_warnings`